### PR TITLE
Change config to use unstructured

### PR DESCRIPTION
**Pull request by RamLavi**
This PR is changes config operations to support both v1, v1alpha1 NetworkAddonsConfig

- **refactor operations to accept unstructured**
Currently, functions that create/update/remove NetworkAddonsConfig type
only use the currently stored NetworkAddonsConfig version:v1
This is not an issue during upgrade tests since we're not upgrading
the crd itself, so we always support v1.
Let's support both APIs by using unstructred objects
when calling k8s client.

- **update e2e tests to use new config changes**
this commit introduces the change in config API to
all the relevant e2e tests

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/operations/operations.go [hunk 1]
```diff
@@ -8,74 +8,117 @@ import (
...
+	cnaoV1alpha1 := cnaov1.NetworkAddonsConfig{}
+	switch gvk {
+	case cnaoV1.GroupVersionKind():
+		return &convertToConfigV1(config).Status
+	case cnaoV1alpha1.GroupVersionKind():
+		return &convertToConfigV1alpha1(config).Status
+	}
+
+	Fail(fmt.Sprintf("gvk %v not supported", gvk))
+	return nil
```
Comment (qinqon) [inline test/operations/operations.go:97]: We cannot retrieve status from unstructured ?

Comment (RamLavi) [reply test/operations/operations.go:97]: `framework.Global.Client.Status().` doesn't support `Get`,  only `Update\Patch`.
I did not find a way to get the status object out of the struct

File: test/operations/operations.go [hunk 2]
```diff
@@ -8,74 +8,123 @@ import (
...
-			Name: names.OPERATOR_CONFIG,
-		},
-	}
+	config := &unstructured.Unstructured{}
+	config.SetGroupVersionKind(gvk)
+	config.SetName(names.OPERATOR_CONFIG)
 
 	err := framework.Global.Client.Delete(context.TODO(), config)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to remove the Config")
+	Expect(err).NotTo(HaveOccurred(), "Failed to remove the Config")
```
Comment (qinqon) [inline test/operations/operations.go:73]: This is not related.

**Review by qinqon — CHANGES_REQUESTED** [review]
Just one nit

File: test/operations/operations.go (hunk 2 above)
Comment (RamLavi) [reply test/operations/operations.go]: Done
